util_test.go

```diff
@@ -0,0 +1,161 @@
+// Copyright © 2022 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sdk
+
+import (
+	"testing"
+	"time"
+
+	"github.com/matryer/is"
+)
+
+func TestParseConfig_Simple_Struct(t *testing.T) {
+	is := is.New(t)
+
+	type Person struct {
+		Name string `mapstructure:"person_name"`
+		Age  int
+	}
+
+	input := map[string]string{
+		"person_name": "meroxa",
+		"age":         "91",
+	}
+	want := Person{
+		Name: "meroxa",
+		Age:  91,
+	}
+
+	var got Person
+	err := Util.ParseConfig(input, &got)
+	is.NoErr(err)
+	is.Equal(want, got)
+}
+
+func TestParseConfig_Embedded_Struct(t *testing.T) {
+	is := is.New(t)
+
+	type Family struct {
+		LastName string
+	}
+	type Location struct {
+		City string
+	}
+	type Person struct {
+		Family    `mapstructure:",squash"`
+		Location  `mapstructure:",squash"`
+		FirstName string
+	}
+
+	input := map[string]string{
+		"FirstName": "conduit",
+		"lastname":  "meroxa",
+		"City":      "San Francisco",
+	}
+	want := Person{
+		Family:    Family{LastName: "meroxa"},
+		Location:  Location{City: "San Francisco"},
+		FirstName: "conduit",
+	}
+
+	var got Person
+	err := Util.ParseConfig(input, &got)
+	is.NoErr(err)
+	is.Equal(want, got)
+}
+
+func TestParseConfig_All_Types(t *testing.T) {
+	is := is.New(t)
+	type Config struct {
```

Comment:
How about a slice? I'd expect to be able to parse a comma separated string into `[]string` (e.g. `a,b,c`). Bonus points if we can parse into `[]T` where `T` is any base type (that might be a bit harder though).

Another thing would be maps. For example, do we want to parse something like `key1:val1,key2:val2` into `map[string]string{"key1":"val1","key2":"val2"}`?

Reply:
@lovromazgon I don't think we ever had a config value as a map so far.. do you think we should support that too?

Reply:
We have a case in the generator where we do this ([link to a test](https://github.com/ConduitIO/conduit-connector-generator/blob/884dfe8d4dc4ed0c631830c7165d0c1934ab8254/config_test.go#L38)), but to be fair it's the only use case and I'm not sure there will be many more, so a map is not essential. A slice on the other hand would be useful IMO.

Reply:
Cool, I see the hook you added works for `[]T`, that's pretty neat! I thought it would only work for `[]string`.